apps/admin-portal/src/pages/applications.tsx

```diff
@@ -114,21 +139,89 @@ export const ApplicationsPage = (): JSX.Element => {
      */
     const handleApplicationFilter = (query: string): void => {
         setSearchQuery(query);
+        getAppLists(listItemLimit, listOffset, query);
     };
 
-    const handlePaginationChange = (event: React.MouseEvent<HTMLAnchorElement>, data: PaginationProps) => {
+    /**
+     * Handles the pagination change.
+     *
+     * @param {React.MouseEvent<HTMLAnchorElement>} event - Mouse event.
+     * @param {PaginationProps} data - Pagination component data.
+     */
+    const handlePaginationChange = (event: React.MouseEvent<HTMLAnchorElement>, data: PaginationProps): void => {
         setListOffset((data.activePage as number - 1) * listItemLimit);
     };
 
-    const handleItemsPerPageDropdownChange = (event: React.MouseEvent<HTMLAnchorElement>, data: DropdownProps) => {
+    /**
+     * Handles per page dropdown page.
+     *
+     * @param {React.MouseEvent<HTMLAnchorElement>} event - Mouse event.
+     * @param {DropdownProps} data - Dropdown data.
+     */
+    const handleItemsPerPageDropdownChange = (event: React.MouseEvent<HTMLAnchorElement>, data: DropdownProps): void => {
         setListItemLimit(data.value as number);
     };
 
     /**
      * Handles application delete action.
      */
     const handleApplicationDelete = (): void => {
-        getAppLists(listItemLimit, listOffset);
+        getAppLists(listItemLimit, listOffset, null);
+    };
+
+    /**
+     * Handles the `onSearchQueryClear` callback action.
+     */
+    const handleSearchQueryClear = (): void => {
+        setSearchQuery("");
+        getAppLists(listItemLimit, listOffset, null);
+    };
+
+    /**
+     * Resolve the relevant placeholder.
+     *
+     * @return {JSX.Element}
+     */
+    const showPlaceholders = (): JSX.Element => {
```

Comment:
Replace `JSX.Element` since `ReactElement` is already used in the file.

Reply:
Fixed with commit `e941ae3`